src/kernel/arch/aarch64/interrupts.zig

```diff
@@ -73,3 +175,66 @@ export fn fastIrqHandler() callconv(.Naked) noreturn {
     log.logError("fast irq at instruction 0x{X}\n", .{interruptedInstruction()});
     while (true) {}
 }
+
+inline fn lr() usize {
+    var word = asm ("mov %[word], lr"
+        : [word] "=r" (-> usize)
+    );
+    return word;
+}
+
+inline fn sp() usize {
+    var word = asm ("mov %[word], sp"
+        : [word] "=r" (-> usize)
+    );
+    return word;
+}
+
+fn cpsr() usize {
+    var word = asm ("mrs %[word], cpsr"
+        : [word] "=r" (-> usize)
+    );
+    return word;
+}
+
+fn spsr() usize {
+    var word = asm ("mrs %[word], spsr"
+        : [word] "=r" (-> usize)
+    );
+    return word;
+}
+
+fn sctlr() usize {
+    var word = asm ("mrc p15, 0, %[word], c1, c0, 0"
+        : [word] "=r" (-> usize)
+    );
+    return word;
+}
+
+fn scr() u32 {
+    var word = asm ("mrc p15, 0, %[word], c1, c1, 0"
+        : [word] "=r" (-> usize)
+    );
+    return word;
+}
+
+inline fn mrs(comptime register_name: []const u8) u64 {
+    const word = asm ("mrs %[word], " ++ register_name
+        : [word] "=r" (-> u64)
+    );
+    return word;
+}
+
+pub inline fn mrsEl(comptime register_name: []const u8, comptime exception_level: u32) u64 {
+    const exception_level_string = switch (exception_level) {
+        1 => "1",
+        2 => "2",
+        3 => "3",
+        else => unreachable,
+    };
+    return mrs(register_name ++ exception_level_string);
+}
+
+pub inline fn currentEl() u2 {
```

Comment:
This function name could be a bit more descriptive, like `currentExceptionLevel`.

Reply:
I was thinking that myself. I will update it.

Reply:
Done.